### PR TITLE
Feature/separate hydrology

### PR DESCRIPTION
**Pull request by vgro**
I moved the hydrology (soil moisture and runoff) out of the `abiotic_simple_model` to a new model (`hydrology_model`)  to make it easier to plug in different (external) hydrology models that might have different levels of complexity and different time steps; for example HydroPy, SPLASH, WSIMOD. The functions are the same as before, just in a separate model. This means that the `hydrology` initializes the `soil_moisture` variable which the `soil` model depends on, so has to come before that.

In the course of this, I renamed the `simple_regression` module to `microclimate` as this is more descriptive of what is going on there.

There are still some features missing: the vertical (downward) flow, the subsurface flow, the horizontal flow between grid cells, the stream flow and the extraction of water by plant roots (=evapotranspiration). This will be implemented step by step after discussion with group members as it requires some thinking about the order of processes.

Question: The equation at the end of `hydrology_model` does not show in the online documentation, any suggestions why this could be?

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: virtual_rainforest/models/hydrology/hydrology_model.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        soil_moisture_capacity - current_soil_moisture.mean(dim="layers")
+    ) * 1000
+
+    # calculate where precipitation exceeds available capacity
+    surface_runoff_cells = precipitation_surface.where(
+        precipitation_surface > available_capacity
+    )
+    # calculate runoff
+    surface_runoff = (
+        DataArray(surface_runoff_cells.data - available_capacity.data / 1000)
```
Comment (jacobcook1995) [inline virtual_rainforest/models/hydrology/hydrology_model.py:248]: Where does this `/1000` come from? (and the `*1000` above)

**Review by jacobcook1995 — APPROVED** [review]
Looks like a sensible restructure to me! I've added a few minor comments, but nothing major really

File: docs/source/api/hydrology/hydrology_model.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    format_version: 0.13
+    jupytext_version: 1.13.8
+kernelspec:
+  display_name: vr_python3
+  language: python
+  name: vr_python3
+---
+
+<!-- markdownlint-disable-next-line  MD013 -->
+# API documentation for the {mod}`~virtual_rainforest.models.hydrology.hydrology_model` module
```
[the remark this answers is not in the document]
Comment (vgro) [reply docs/source/api/hydrology/hydrology_model.md:18]: that worked, thanks :-)

**Comment by vgro** [comment]
> This looks really good - and definitely the refactoring makes sense now that the codebase an d potential use is growing. Excellent use of some of xarray capabilities.
> 
> I'm not sure on the question about the documentation. What equation does not show?

Thank you for the positive feedback.
We solved the issue with the documentation, @jacobcook1995 came across a similar problem before and had a solution.

**Review by davidorme — APPROVED** [review]
LGTM too.

**Pull request merged**